packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -263,14 +275,28 @@ async function askUserForFilepathConfirmation(
 
 export async function resolveFoamTemplateVariables(
   templateText: string,
-  extraVariablesToResolve: Set<string> = new Set()
+  extraVariablesToResolve: Set<string> = new Set(),
+  givenValues: Map<string, string> = new Map()
 ): Promise<[Map<string, string>, string]> {
-  const givenValues = new Map<string, string>();
   const variables = findFoamVariables(templateText.toString()).concat(
     ...extraVariablesToResolve
   );
   const uniqVariables = [...new Set(variables)];
 
+  const missingFoamVariables = [...givenValues.keys()].filter(
+    x => uniqVariables.indexOf(x) < 0
+  );
+  if (
+    missingFoamVariables.length > 0 &&
+    missingFoamVariables.includes('FOAM_SELECTED_TEXT')
```

Comment:
here we are not checking that an actual value exists for `FOAM_SELECTED_TEXT`.
Normally in templates a non-existing variable is simply left there untouched, in our case we would want the text to be empty.
So, we either:
- default `FOAM_SELECTED_TEXT` to the empty string (my personal favorite option, but I am open to disagreements)
- only append `FOAM_SELECTED_TEXT` in the template if it's also defined in the provided variables

what do you think